src/transactions/FeeBumpTransactionFrame.cpp

```diff
@@ -314,38 +314,51 @@ FeeBumpTransactionFrame::commonValidPreSeqNum(
         txResult.setError(txINSUFFICIENT_FEE);
         return std::nullopt;
     }
-    // While in theory it should be possible to bump a Soroban
-    // transaction with negative inclusion fee (this is unavoidable
-    // when Soroban resource fee exceeds uint32), we still won't
-    // consider the inner transaction valid. So we return early here
-    // in order to have `bigMultiply` below not crash.
-    if (mInnerTx->getInclusionFee() < 0)
-    {
-        txResult.setError(txFEE_BUMP_INNER_FAILED);
-        return std::nullopt;
-    }
-    auto const& lh = header.current();
-    // Make sure that fee bump is actually happening, i.e. that the
-    // inclusion fee per operation in this envelope is higher than
-    // the one in the inner envelope.
-    uint128_t v1 =
-        bigMultiply(getInclusionFee(), getMinInclusionFee(*mInnerTx, lh));
-    uint128_t v2 =
-        bigMultiply(mInnerTx->getInclusionFee(), getMinInclusionFee(*this, lh));
-    if (v1 < v2)
-    {
-        int64_t feeNeeded = 0;
-        if (!bigDivide128(feeNeeded, v2, getMinInclusionFee(*mInnerTx, lh),
-                          Rounding::ROUND_UP))
+    auto innerInclusionFee = mInnerTx->getInclusionFee();
+    if (innerInclusionFee > 0)
```

Comment:
Should this be `innerInclusionFee >= 0`? The check for the old code was `mInnerTx->getInclusionFee() < 0`.

Reply:
Good catch, fixed and added a test for 0 inclusion fee.